lora/lora.py

```diff
@@ -205,15 +211,18 @@ def iterate_batches(dset, tokenizer, batch_size, train=False):
 
         # Collect batches from dataset
         for i in range(0, len(indices) - batch_size + 1, batch_size):
-            # Encode batch
+            # Encode batch and truncate if necessary
             batch = [
-                tokenizer.encode(dset[indices[i + j]], eos=True)
+                tokenizer.encode(dset[indices[i + j]], eos=True)[:max_seq_length]
                 for j in range(batch_size)
             ]
             lengths = [len(x) for x in batch]
 
+            # Ensure max length does not exceed max_seq_length
+            max_len = min(max(lengths), max_seq_length)
+            
             # Pad to the max length
-            batch_arr = np.zeros((batch_size, max(lengths)), np.int32)
+            batch_arr = np.zeros((batch_size, max_len), np.int32)
```

Comment:
I don't think we should do this. If someone has just a few long documents, then this could silently drop almost all of their data.

My suggestion instead is to pre split data as a preprocessing step prior to making the json files. The added benefit of doing it that way is split points might be applications specific.

One thing we could do here is add a warning, like if `max(lengths) > 2048` print a warning suggesting the user pre split their data to save memory.

Reply:
Sure, I would add this to the code.